st2common/st2common/services/packs.py

```diff
@@ -215,3 +219,71 @@ def search_pack_index(
                 break
 
     return list(itertools.chain.from_iterable(matches))
+
+
+def delete_action_files_from_pack(pack_name, entry_point, metadata_file):
+    """
+    Prepares the path for entry_point file and metadata file of action and
+    deletes them from disk.
+    """
+
+    pack_base_path = get_pack_base_path(pack_name=pack_name)
+    action_entrypoint_file_path = os.path.join(pack_base_path, "actions", entry_point)
+    action_metadata_file_path = os.path.join(pack_base_path, metadata_file)
+
+    if os.path.exists(action_entrypoint_file_path):
+        try:
+            os.remove(action_entrypoint_file_path)
+        except PermissionError:
+            LOG.error(
+                'No permission to delete the "%s" file',
+                action_entrypoint_file_path,
+            )
+            msg = 'No permission to delete "{0}" file from disk'.format(
+                action_entrypoint_file_path
+            )
+            raise PermissionError(msg)
+        except Exception as e:
+            LOG.error(
+                'Unable to delete "%s" file. Exception was "%s"',
+                action_entrypoint_file_path,
+                e,
+            )
+            msg = 'The action file "{0}" could not be removed from disk, please check the logs or ask your StackStorm administrator to check and delete the actions files manually'.format(
+                action_entrypoint_file_path
+            )
+            raise ResourceDiskFilesRemovalError(msg)
+    else:
+        LOG.warning(
+            'The action entry point file "%s" does not exists on disk.',
+            action_entrypoint_file_path,
+        )
+
+    if os.path.exists(action_metadata_file_path):
+        try:
+            os.remove(action_metadata_file_path)
+        except PermissionError:
+            LOG.error(
+                'No permission to delete the "%s" file',
+                action_metadata_file_path,
+            )
+            msg = 'No permission to delete "{0}" file from disk'.format(
+                action_metadata_file_path
+            )
+            raise PermissionError(msg)
+        except Exception as e:
+            LOG.error(
+                'Could not delete "%s" file. Exception was "%s"',
+                action_metadata_file_path,
+                e,
+            )
+            if os.path.isfile(action_metadata_file_path):
```

Comment:
This check is probably redundant, since if `os.remove()` fails it will throw some sort of exception.